src/core/types.ts

```diff
@@ -108,5 +108,4 @@ export interface DepGraphInternal extends DepGraph {
   getPkgNodeIds(pkg: Pkg): string[];
   getNodeDepsNodeIds(nodeId: string): string[];
   getNodeParentsNodeIds(nodeId: string): string[];
-  hasCycles(): boolean;
```

Comment:
This is a breaking change, and we must release it as such (v2). Even without this change I'd argue that this feature is worth a new major version.

Reply:
is it breaking? it's was only on `DepGraphInternal`

Reply:
not exposed in https://github.com/snyk/dep-graph/blob/master/src/index.ts#L10

Reply:
ah yes! I'd still argue that this is worth a new major version due to the risk it introduces and the potential performance impact. Wouldn't want to catch anyone by surprise in a minor automated dep bump; would rather people opted-in.

Reply:
👍 if we do this would like to slide-in another small change. Removing `getDepPkgs()` and providing the same functionality via a param to `getPkgs()`. reason being: avoiding duplicate meaning to the word "Dep", as in "direct deps" vs "all deps", due to new methods I would like to expose with ttps://github.com/snyk/dep-graph/pull/47